### PR TITLE
Getting Started with Optical Character Recognition

### LINKED ISSUE #5697: [Machine Learning] Getting Started with Optical Character Recognition (OCR)
#### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. 

Getting Started with Optical Character Recognition (OCR)

#### Proposed article introduction
*We're looking for the first 2-3 paragraphs of your article that appropriately summarize what your article will be about.*
We will use this writing snippet to help us assess the overall quality before approval so ensure it has been proofread, spellchecked and grammarchecked.

Optical character recognition (OCR) is a technology that is used to turn scanned images and other visuals into text. This turns paper-based documents into editable and searchable digital mediums and enables the development of automated systems. This tutorial will leverage EasyOCR, a Python package that makes it easy to use this technology in the simplest way possible.

#### Key takeaways 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?
Use this area to get your ideas down on the bulk of your article or tutorial.

By the end of this tutorial, a reader will be able to understand:
- What is Optical Character Recognition?
- How to set up and install EasyOCR. EasyOCR is a package that makes it easy to perform OCR.
- How to use EasyOCR to extract text from different images.
- How to visualize results using the OpenCV library.

#### Article quality
How is your article unique? Tell us what makes your approach different from similar articles that have published on the same topic?
Is yours more in-depth? Does it cover additional topics? Do you provide handy tips or anecdotal advice?

My goal is to make people understand AI in the most simplest way. This tutorial will take our readers through how to get started using EasyOCR  which is a package that makes it easy to use OCR. This will be accomplished with only a few lines of code.

### PR DESCRIPTION
**Pull request by lcheptoo**
This closes #5697

## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
The following code shows plagiarism. This includes variable names. Please correct it.

```python

img = cv2.imread("image-one.jpg")

spacer = 100

for detection in results:

top_left = tuple(detection[0][0])

bottom_right = tuple(detection[0][2])

text = detection[1]

img = cv2.rectangle(img,top_left,bottom_right,(0,255,0),3)

img = cv2.putText(img,text,(20,spacer), font, 0.5,(0,255,0),2,cv2.LINE_AA)

spacer+=15


plt.imshow(img)

plt.show()

```

**Comment by collins-a** [comment]
> The following code shows plagiarism. This includes variable names. Please correct it.
> 
> ```python
> img = cv2.imread("image-one.jpg")
> 
> spacer = 100
> 
> for detection in results:
> 
> top_left = tuple(detection[0][0])
> 
> bottom_right = tuple(detection[0][2])
> 
> text = detection[1]
> 
> img = cv2.rectangle(img,top_left,bottom_right,(0,255,0),3)
> 
> img = cv2.putText(img,text,(20,spacer), font, 0.5,(0,255,0),2,cv2.LINE_AA)
> 
> spacer+=15
> 
> 
> plt.imshow(img)
> 
> plt.show()
> ```

@lcheptoo just following up to ensure you saw the above comments on the areas of your code that need changing before I review your article.

**Comment by lcheptoo** [comment]
> > The following code shows plagiarism. This includes variable names. Please correct it.
> > ```python
> > img = cv2.imread("image-one.jpg")
> > 
> > spacer = 100
> > 
> > for detection in results:
> > 
> > top_left = tuple(detection[0][0])
> > 
> > bottom_right = tuple(detection[0][2])
> > 
> > text = detection[1]
> > 
> > img = cv2.rectangle(img,top_left,bottom_right,(0,255,0),3)
> > 
> > img = cv2.putText(img,text,(20,spacer), font, 0.5,(0,255,0),2,cv2.LINE_AA)
> > 
> > spacer+=15
> > 
> > 
> > plt.imshow(img)
> > 
> > plt.show()
> > ```
> 
> @lcheptoo just following up to ensure you saw the above comments on the areas of your code that need changing before I review your article.

@collins-a done.

File: content/articles/getting-started-with-optical-character-recognition/index.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    image = cv2.rectangle(image,top_left_detection,bottom_right_detection,(255,0,0),3)
+    image = cv2.putText(image,text_detection,(20,spacer), font, 0.5,(255,0,0),2,cv2.LINE_AA)
+    spacer+=20
+    
+plt.imshow(image)
+plt.show()
+```
+Result:
+
+![Multiple lines](/engineering-education/getting-started-with-optical-character-recognition/multiple-lines.png)
```
Comment (collins-a) [inline content/articles/getting-started-with-optical-character-recognition/index.md:157]: This image is missing from the article files

File: content/articles/getting-started-with-optical-character-recognition/index.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    bottom_right_detection = tuple(detection[0][2])
+    text_detection = detection[1]
+    image = cv2.rectangle(image,top_left_detection,bottom_right_detection,(255,0,0),3)
+    image = cv2.putText(image,text_detection,(20,spacer), font, 0.5,(255,0,0),2,cv2.LINE_AA)
+    spacer+=20
+    
+plt.imshow(image)
+plt.show()
+```
+Result:
```
Comment (collins-a) [inline content/articles/getting-started-with-optical-character-recognition/index.md:155]: Running the code block for **handling multiple lines** yields the following error

```python
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
<ipython-input-22-61b6150f70c9> in <module>()
      5     bottom_right = tuple(detection[0][2])
      6     text = detection[1]
----> 7     img = cv2.rectangle(img,top_left,bottom_right,(0,255,0),3)
      8     img = cv2.putText(img,text,(20,spacer), font, 0.5,(0,255,0),2,cv2.LINE_AA)
      9     spacer+=15

error: OpenCV(4.5.4) :-1: error: (-5:Bad argument) in function 'rectangle'
> Overload resolution failed:
>  - Can't parse 'pt1'. Sequence item with index 0 has a wrong type
>  - Can't parse 'pt1'. Sequence item with index 0 has a wrong type
>  - Can't parse 'rec'. Expected sequence length 4, got 2
>  - Can't parse 'rec'. Expected sequence length 4, got 2
```

**Review by collins-a — CHANGES_REQUESTED** [review]
Near flawless tutorial on OCR @lcheptoo. A couple of changes and we're set for the final review.

**Comment by lcheptoo** [comment]
> Near flawless tutorial on OCR @lcheptoo. A couple of changes and we're set for the final review.

@collins-a done.

**Review by collins-a — APPROVED** [review]

**Comment by ahmadmardeni1** [comment]
Great work team. @lcheptoo @collins-a

**Pull request merged**